src/execution/expression_executor/execute_function.cpp

```diff
@@ -3,21 +3,28 @@
 
 namespace duckdb {
 
+struct FunctionExpressionState : public ExpressionState {
+	FunctionExpressionState(Expression &expr, ExpressionExecutorState &root) : ExpressionState(expr, root) {
+	}
+
+	DataChunk arguments;
+};
 unique_ptr<ExpressionState> ExpressionExecutor::InitializeState(BoundFunctionExpression &expr,
                                                                 ExpressionExecutorState &root) {
-	auto result = make_unique<ExpressionState>(expr, root);
+	auto result = make_unique<FunctionExpressionState>(expr, root);
 	for (auto &child : expr.children) {
 		result->AddChild(child.get());
 	}
 	result->Finalize();
-	return result;
+	result->arguments.InitializeEmpty(result->types);
```

Comment:
Seems to be a minor bug here, this needs to be:

```
if (result->types.size() > 0) {
    result->arguments.InitializeEmpty(result->types);
}
```

To avoid triggering an assertion on functions without parameters (e.g. `random()`).